src/OrchardCore/OrchardCore.Abstractions/Caching/Distributed/IDistributedCache.cs

```diff
@@ -0,0 +1,20 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Distributed;
+
+namespace OrchardCore.Caching.Distributed
+{
+    public interface IDistributedCache
```

Comment:
This is kind of the same as
https://github.com/aspnet/Caching/blob/master/src/Microsoft.Extensions.Caching.Abstractions/IDistributedCache.cs

Reply:
Yes, the problem is that if you use the same interface and AddStackExchangeRedis() you either have to extract/wrap the dependency it creates OR stop using that call and recreate it.

I actually thought of a solution last night.. in the constructor for DocumentManager it could ask for a MS IDistributedCache and then wrap it as long as I changed the RedisDistributedCache into a more generic version.